pkg/rekor/identity.go

```diff
@@ -105,8 +109,217 @@ func (e *IdentityEntry) String() string {
 	return strings.Join(parts, " ")
 }
 
+func renderFulcioOIDMatchers(e extensions.FulcioExtensions) ([]OIDMatcher, error) {
```

Comment:
Can we add a test for `renderFulcioOIDMatchers`, which fully populates a `FulcioExtensions` struct and checks its output?

Another question, should this be a function on the `FulcioExtensions` struct rather than its own method?

Reply:
working on adding a test- this method is kind of weird in terms of placement, since ideally it should be a method on the `FulcioExtensions` struct and that would also require it to be inside `pkg/fulcio/extensions.go` but it also depends on `OIDMatcher`, which then causes an import loop. I considered duplicating the OIDMatcher type, which would solve some of these issues but also creates a bunch of new ones and generally seems pretty clunky and not preferable- an option I just came up with would involve moving this function back into `extensions.go` and having it output a map of `ObjectIdentifier` (as string, which would then need to be parsed) to `[]string` and writing another helper in `identity.go` that parses the map. Let me know if that would be preferable to what we currently have or if there's another option you would like- honestly a lot of these options all seem to have some various drawbacks

Reply:
For breaking the circular dependency, I'm thinking we could move `OIDMatcher` to its own package/file and have both `identity.go` and `extensions.go` depend on that? Or I'm missing something

Reply:
Agreed, that sounds like a clean solution.

Reply:
will update accordingly